templates/cluster-template-persistent-example.yaml

```diff
@@ -0,0 +1,223 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks: ["${POD_NETWORK_CIDR:=192.168.0.0/16}"]
+    services:
+      cidrBlocks: ["${SERVICE_CIDR:=10.96.0.0/12}"]
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: VirtinkCluster
+    name: "${CLUSTER_NAME}"
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+    kind: KubeadmControlPlane
+    name: "${CLUSTER_NAME}-cp"
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: VirtinkCluster
+metadata:
+  name: "${CLUSTER_NAME}"
+spec:
+  controlPlaneServiceType: "${VIRTINK_CONTROL_PLANE_SERVICE_TYPE:=NodePort}"
+  # This is am example template for host cluster which use Calico CNI.
+  # When use this example template, firstly make sure your host cluster CNI support use specific IP for pod.
+  # 1. replace 'addresses' to the actual available IPs provided by your host cluster CNI,
```

Comment:
Make these fields configurable by OS env and add it to documentation `README.md` ? Same with `datavolume.spec.url`

Reply:
> Make these fields configurable by OS env and add it to documentation `README.md` ? Same with `datavolume.spec.url`

at first, I thought this template is just an example, user need to download and modify it. because `addresses` is an array, and `annotations` is a map, but it seems clusterctl template don't support replace array or map by env variable. 

for using it directly with env variable need change `addresses` and `annotations` to simple string type.